### DISCUSSION TITLE
Determining which heating source is being used?

### DISCUSSION THREAD

**Discussion by joeshaw** [PeteRager/lennoxs30#351] [Q&A] [answered]
Hi,

I have an S40 with a dual-fuel heating setup: a gas furnace and an electric heat pump.  The way the system is configured, the HP is the main heating source and the furnace is the auxiliary heat.  I don't believe it is considered "emergency heat", if that makes a difference.

From the HA perspective, I can see when the system is heating but it doesn't seem like I can tell _which_ source of heat it is using.  On the S40 and the mobile app, when it is using the furnace in addition to saying "Heating" it says "Using aux heat" (or something like that, I don't recall the exact wording).

It seems like perhaps the `sensor.<system>_alert` entity gets set to "info" when the aux heat kicks on, but I'm not sure that correlation is accurate, and it doesn't seem like high enough fidelity info to rely upon it.

Am I missing a better way to solve this?

*Edit*: I am using HA 2024.12.1 and version 2024.11.0 of this integration.

**Comment by PeteRager** [comment]
Have you looked at the aux attribute of the climate entity?

**Reply by joeshaw** [reply, marked as answer]
I had not, and yes, that will work nicely.  I will create a template binary sensor for this.  Thank you.

I think I was thrown off by the existence of the ambient lockout binary sensors.  This is perhaps a newbie question, but how do you decide when to create a binary sensor versus just exposing things via attributes on the main climate entity?

**Reply by PeteRager** [reply]
I try to keep data packaged in a similar way that it arrives from Lennox.  In their model, the aux is a property of the zone status and gets sent with the zone data; same with other attributes like SSR, so I put those in the climate entity rather than creating a lot of sensors and cluttering up the entity list. Templates make it easy to create a sensor if you want one.

https://github.com/PeteRager/lennoxs30api/blob/master/tests/messages/system_04_furn_ac_zoning_zones.json
